core/src/main/java/tc/oc/pgm/death/DeathMessageMatchModule.java

```diff
@@ -41,25 +42,35 @@ public void handleDeathBroadcast(MatchPlayerDeathEvent event) {
     Component message = builder.getMessage().color(NamedTextColor.GRAY);
 
     for (MatchPlayer viewer : event.getMatch().getPlayers()) {
+      boolean involved = event.isInvolved(viewer) || event.isInvolved(viewer.getSpectatorTarget());
+      boolean isStaff = event.isTeamKill() && viewer.getBukkit().hasPermission(Permissions.STAFF);
+
       switch (viewer.getSettings().getValue(SettingKey.DEATH)) {
         case DEATH_OWN:
-          if (event.isInvolved(viewer) || event.isInvolved(viewer.getSpectatorTarget())) {
+          if (involved) {
             viewer.sendMessage(message);
-          } else if (event.isTeamKill() && viewer.getBukkit().hasPermission(Permissions.STAFF)) {
+          } else if (isStaff) {
             viewer.sendMessage(message.decoration(TextDecoration.ITALIC, true));
           }
           break;
         case DEATH_FRIENDS:
           if (event.isInvolved(viewer)) {
             viewer.sendMessage(message.decoration(TextDecoration.BOLD, true));
-          } else if (isFriendInvolved(viewer.getBukkit(), event)) {
-            viewer.sendMessage(message);
-          } else if (event.isTeamKill() && viewer.getBukkit().hasPermission(Permissions.STAFF)) {
-            viewer.sendMessage(message.decoration(TextDecoration.ITALIC, true));
+          } else if (isFriendInvolved(viewer.getBukkit(), event) || isStaff) {
+            viewer.sendMessage(message.decoration(TextDecoration.ITALIC, isStaff));
+          }
+          break;
+        case DEATH_SQUAD:
+          if (involved) {
+            viewer.sendMessage(message.decoration(TextDecoration.BOLD, true));
+          } else if (isSquadInvolved(viewer.getBukkit(), event)
+              || isFriendInvolved(viewer.getBukkit(), event)
+              || isStaff) {
+            viewer.sendMessage(message.decoration(TextDecoration.ITALIC, isStaff));
           }
           break;
         case DEATH_ALL:
-          if (event.isInvolved(viewer) || event.isInvolved(viewer.getSpectatorTarget())) {
+          if (involved) {
             viewer.sendMessage(message.decoration(TextDecoration.BOLD, true));
           } else {
             viewer.sendMessage(message);
```

Comment:
This switch was already a monstruosity before, and it just gets worse.

Maybe changing this to a switch expression that returns true/false for if the msg should show, then just a simple check after to decide if bold (own) or italic (staff & teamgrief) is applied ontop.


```java
boolean show = involved || isStaff || switch (viewer.getSettings().getValue(SettingKey.DEATH)) {
        case DEATH_OWN -> false;
        case DEATH_FRIENDS -> isFriendInvolved(...);
        case DEATH_SQUAD -> isFriendInvolved(...) ||  isSquadInvolved(...);
        case DEALTH_ALL -> true;
}
if (show) {
  if (involved) message = message.decoration(TextDecoration.BOLD, true);
  else if (isStaff) message = message.decoration(TextDecoration.ITALIC, true);
  viewer.sendMessage(message);
}
```

Reply:
Great suggestion! I wasn't sure how much I could simplify before, but I'm glad we were able to improve it even more by getting rid of that awful-looking switch statement. Let me know if there's anything else you'd like to see changed. Thanks 👍